### ISSUE TITLE
Columns:: generatePageHook expects global \PageRegular

### ISSUE THREAD

**Issue by DanielSchwiperich** [madeyourday/contao-rocksolid-columns#14]
The hook expects a `\PageRegular` instead of a `Contao\PageRegular` which leads to something like
```
Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Type error: Argument 3 passed to MadeYourDay\\RockSolidColumns\\Columns::generatePageHook() must be an instance of PageRegular, instance of Ctsmedia\\Phpbb\\BridgeBundle\\PageType\\Forum
```

when a object is passed which extends `Contao\PageRegular` like for example our bridge forum type
https://github.com/ctsmedia/contao-phpbb-bridge-bundle/blob/master/src/PageType/Forum.php#L27

If I recall correctly Contao did this so that you could rewrite classes and it's only legacy stuff today. 
Not sure if your module needs still to be compatible with this approach. 
If not it should be changed for PageModel, LayoutModel and Database in this class too. 

I could provide a PR for it, if you like

**Comment by ausi** [comment]
Fixed in the master branch, could you please try `"madeyourday/contao-rocksolid-columns": "dev-master as 2.0.1"`?

> If not it should be changed for PageModel, LayoutModel and Database in this class too.

I don’t think that the call to `\Database::getInstance()` has do be changed. If the contao framework is booted, `\Database` should always be available. It should only be relevant for parameter type hints.

**Comment by DanielSchwiperich** [comment]
thanks @ausi, looks good, will test it tomorrow

**Comment by DanielSchwiperich** [comment]
Fix erfolgreich getestet. Danke @ausi

**Comment by ausi** [comment]
Released as Version 2.0.1.
